packages/synapse-sdk/src/synapse.ts

```diff
@@ -166,6 +168,9 @@ export class Synapse {
       pieceRetriever = new FilBeamRetriever(baseRetriever, network)
     }
 
+    // Create FilBeamService
+    const filbeamService = await FilBeamService.create(provider)
```

Comment:
You should change this to just take `network`, not `provider` because that's all you're using it for. Then it doesn't need to be async, you can just use a constructor instead of the async factory, and you don't have yet another `getNetworkType` call in the Synapse initialisation path (it should be cached, but we don't have a guarantee of that). Nice and fast to get this set up.